setup/gcp/installer/forseti_server_installer.py

```diff
@@ -0,0 +1,339 @@
+
+        self.should_setup_explain()
+        self.determine_access_target()
+        self.should_enable_write_access()
+        self.format_gsuite_service_acct_id()
+        self.inform_access_on_target()
+
+        enable_apis(self.config.dry_run)
+        self.gsuite_service_account = create_reuse_service_acct(
+            'gsuite_service_account',
+            self.gsuite_service_account,
+            self.config.advanced_mode,
+            self.config.dry_run)
+        self.get_email_settings()
+
+    def deploy(self, deploy_tpl_path, conf_file_path, bucket_name):
+        """Deploy Forseti using the deployment template.
+        Grant access to service account.
+
+        Args:
+            deploy_tpl_path (str): Deployment template path
+            conf_file_path (str): Configuration file path
+            bucket_name (str): Name of the GCS bucket
+
+        Returns:
+            bool: Whether or not the deployment was successful
+            str: Deployment name
+        """
+        success, deployment_name = super(ForsetiServerInstaller, self).deploy(
+            deploy_tpl_path, conf_file_path, bucket_name)
+
+        if success:
+            self.has_roles_script = grant_server_svc_acct_roles(
+                self.enable_write_access,
+                self.access_target,
+                self.target_id,
+                self.project_id,
+                self.gsuite_service_account,
+                self.gcp_service_account,
+                self.user_can_grant_roles)
+
+            # Copy the rule directory to the GCS bucket
+            copy_file_to_destination(
+                RULES_DIR_PATH, bucket_name,
+                is_directory=True, dry_run=self.config.dry_run)
+
+            instance_name = '{}-vm'.format(deployment_name)
+            self.wait_until_vm_initialized(instance_name)
+
+            # Create firewall rule to block out all the ingress traffic
+            create_firewall_rule(
+                self.format_firewall_rule_name('forseti-server-deny-all'),
+                [self.gcp_service_account],
+                FirewallRuleAction.DENY,
+                ['icmp', 'udp', 'tcp'],
+                FirewallRuleDirection.INGRESS,
+                1)
+
+            # Create firewall rule to allow only port tcp:50051
+            create_firewall_rule(
+                self.format_firewall_rule_name('forseti-server-allow-grpc'),
+                [self.gcp_service_account],
+                FirewallRuleAction.ALLOW,
+                ['tcp:50051'],
+                FirewallRuleDirection.INGRESS,
+                0)
+
+        return success, deployment_name
+
+    def format_firewall_rule_name(self, rule_name):
+        """Format firewall rule name.
+
+        Args:
+            rule_name (str): Name of the firewall rule
+
+        Returns:
+            str: Firewall rule name
+        """
+        return '{}-{}'.format(rule_name, self.config.datetimestamp)
+
+    def should_setup_explain(self):
```

Comment:
What does this do now? Explain is enabled by default, and isn't really something that can be disabled today?

Reply:
User is able to disable Explain if they choose to run the installer in advanced mode